plane/src/drone/docker/mod.rs

```diff
@@ -56,11 +54,106 @@ pub struct DockerRuntimeConfig {
     pub cleanup_min_age: Option<Duration>,
 }
 
-#[derive(Clone, Debug)]
+pub type MetricsCallback = Box<dyn Fn(BackendMetricsMessage) + Send + Sync + 'static>;
+
 pub struct DockerRuntime {
     docker: Docker,
     config: DockerRuntimeConfig,
-    _cleanup_handle: Arc<GuardHandle>,
+    metrics_callback: Arc<Mutex<Option<MetricsCallback>>>,
+    events_sender: Sender<TerminateEvent>,
+    _events_loop_handle: GuardHandle,
+    _cleanup_handle: GuardHandle,
+}
+
+async fn events_loop(
+    docker: Docker,
+    metrics_callback: Arc<Mutex<Option<MetricsCallback>>>,
+    event_sender: Sender<TerminateEvent>,
+) {
+    let options = EventsOptions {
+        since: None,
+        until: None,
+        filters: vec![
+            ("type", vec!["container"]),
+            ("event", vec!["die", "stop", "start"]),
+            ("label", vec![PLANE_DOCKER_LABEL]),
+        ]
+        .into_iter()
+        .collect(),
+    };
+    //docker.events(Some(options)).filter_map(|e| {
```

Comment:
I'm guessing this was left in unintentionally?

Reply:
No actually. Removed.